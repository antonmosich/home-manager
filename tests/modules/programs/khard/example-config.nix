{ lib, pkgs, ... }:

{
  programs.khard = {
    enable = true;
    defaultAction = "list";
    settings = {
      general = {
        debug = false;
        editor = [ "vim" "-i" "NONE" ];
        merge_editor = "vimdiff";
      };
      "contact table" = {
        display = "first_name";
        group_by_addressbook = false;
        reverse = false;
        show_nicknames = false;
        show_uids = true;
        show_kinds = false;
        sort = "last_name";
        localize_dates = true;
        preferred_phone_number_type = [ "pref" "cell" "home" ];
        preferred_email_address_type = [ "pref" "work" "home" ];
      };
      vcard = {
        private_objects = [ "Jabber" "Skype" "Twitter" ];
        preferred_version = "3.0";
        search_in_source_files = false;
        skip_unparsable = false;
      };
    };
  };
  accounts.contact.accounts = {
    family = {
      khard.enable = true;
      local.path = "~/.contacts/family/";
    };
    friends = {
      khard.enable = true;
      local.path = "~/.contacts/friends/";
    };
  };

  test.stubs = { khard = { }; };

  nmt.script = ''
    assertFileExists home-files/.config/khard/khard.conf
    assertFileContent home-files/.config/khard/khard.conf ${
      ./example-config.expected
    }
  '';
}
