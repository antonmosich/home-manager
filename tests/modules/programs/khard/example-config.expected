[addressbooks]
[[family]]
path=~/.contacts/family/
[[friends]]
path=~/.contacts/friends/
[contact table]
display=first_name
group_by_addressbook=no
localize_dates=yes
preferred_email_address_type=pref, work, home
preferred_phone_number_type=pref, cell, home
reverse=no
show_kinds=no
show_nicknames=no
show_uids=yes
sort=last_name

[general]
debug=no
default_action=list
editor=vim, -i, NONE
merge_editor=vimdiff

[vcard]
preferred_version=3.0
private_objects=Jabber, Skype, Twitter
search_in_source_files=no
skip_unparsable=no
